apps/verticalhome/js/pin_app_navigation.js

```diff
@@ -0,0 +1,294 @@
+      });
+
+
+      var selected = this.elemList.querySelector('.selected');
+
+      if (selected) {
+        selected.classList.remove('selected')
+      }
+
+      this.selectedElemIndex = 0;
+
+      for (var elemIndex in this.elements) {
+        this.elemList.appendChild(this.elements[elemIndex]);
+      }
+
+      this.elements[this.selectedElemIndex].classList.add('selected');
+      this.elements[this.selectedElemIndex].classList.add('middle');
+      this.elemList.removeAttribute('style');
+      this.middleElem = true;
+
+      var toRemoved = this.elemList.querySelector('.removed');
+      var toRemoveOutFocus = this.elemList.querySelector('.out-focus');
+
+      if (toRemoved) {
+        this.elemList.removeChild(toRemoved);
+      }
+
+      if (toRemoveOutFocus) {
+        toRemoveOutFocus.classList.remove('out-focus');
+      }
+
+      window.removeEventListener('keydown', this);
+      window.addEventListener('keydown', this);
+    },
+
+
+    /**
+     * This function refresh elements by selector
+     * @return {[nothins]} [nothing]
+     */
+    refresh: function refresh() {
+      if (!this.selector) {
+        console.log('Can not work without selector');
+        return;
+      }
+
+      this.elements = Array.prototype.slice.call(document.querySelectorAll(this.selector));
+
+      if (!this.elements && !this.elements.length) {
+        console.log('Can not find any elements by this selector');
+        return;
+      }
+
+      this.elements[this.selectedElemIndex].classList.add('selected');
+        this.elements[this.selectedElemIndex].classList.add('middle');
+        this.middleElem = true;
+    },
+
+
+    /**
+     * Handle event keydown
+     * @param  {[Object]} e [default object]
+     * @return {[nothing]}   [nothing]
+     */
+    handleEvent: function(e) {
+
+      function cycleElements(index) {
+
+        var removed = this.elements.splice(-index, index);
+
+        for (var i = 0; i < removed.length; i++) {
+
+          this.elemList.insertBefore(removed[i], this.elements[0 + i]);
+          this.elements.splice(0 + i, 0, removed[i]);
+
+        }
+
+      };
+
+      if (this.preventDef) {
+        e.preventDefault();
+      }
+      
+      function preVerticalNavigation () {
+        if (this.middleElem) {
+          document.querySelector('.middle').classList.remove('middle');
+          this.middleElem = false;
+        }
+
+
+        var toRemoved = this.elemList.querySelector('.removed');
+        var toRemoveOutFocus = this.elemList.querySelector('.out-focus');
+
+        if (toRemoved) {
+          this.elemList.removeChild(toRemoved);
+        }
+
+        if (toRemoveOutFocus) {
+          toRemoveOutFocus.classList.remove('out-focus');
+        }
+      };
+
+      if (this.preventDef){
+        e.preventDefault();
+      }
+
+
+      switch(e.key) {
+        case 'ArrowUp':
+
+          preVerticalNavigation.call(this);
+          document.getElementById('clock').style.opacity = 0;
```

Comment:
don't set the style directly.  And I think you need some kind of hideClock() method or something more generic for this.